content/cucumber/step-definitions.md

```diff
@@ -253,3 +253,48 @@ Given("I have {int} {color} balls", function (int1, color) {
 });
 ```
 {{% /block %}}
+
+# Abstract Your Steps
+
+It is always a good idea to make sure that you are not writing duplicate/similar step definitions. While there are a number of ways to do so and you can document your step definitions as well, it is always good to make use of **helper methods** to abstract your step definitions and reduce the number of steps.
+
+For example, if you have the following steps used across multiple scenarios :
+
+```
+    Given I go to the home page
+    Given I check the about page of the website 
+    Given I get the contact details
+```
+
+and all of these steps have different step definitions which visit the Home, About and Contact pages, you might be writing _redundant steps_. It is possible that the underlying code for executing each of these test cases is different, **but** it is important to understand that from the point of view of their **behavior**, all of these steps do essentially similar things, ie.
+
+_* Open the Home page_
+_* Open the About page_
+_* Open the Contact page_
+
+As such, you can use abstract helper methods to reduce all these into one step :
+
+    Given I go to the {} page
+
+with the step def :
+
+```java
+@Given("I want to open the {string} page")
+public void i_want_to_open_page(String name) {
+  pageFactory.openPage(name);
+}
+```
+
+And then inside your step definition, use a helper method to determine how the page must be opened and direct your step definition to the correct code for opening that particular page.
```

Comment:
Not part of our Slack discussion, but triggered by your example:
1. There is a balance to consider between reuse of step definitions and helper method, and (over)complicating the implementation. If you get to the point where you need unit tests for your test implementation, it's too complicated ;) In fact, some people prefer step definitions without variables to avoid (more) complicated implementation of step definitions.

2. Personally I use the "helper methods" to "talk to my application", be it the UI / page objects, or the API. This keeps my step definitions "thin" (only providing the glue between Gherkin steps and my code.

Reply:
I did want to portray using the step definitions only as a "glue" between Gherkin and the code. Maybe I wasn't able to convey that properly here. If you could help here, that would be great.

Ah, maybe saying "use a helper method" caused that issue, I should replace that with "Use a factory builder to call the needed code" or something similar for better understanding.

Let me know what you think. You are free to update it as well :)

Reply:
Let me have a look.

Reply:
Ok, so i removed this section and added a more simpler sentence to convey the message properly. Let me know if its better :)